model/coder_junit_result.go

```diff
@@ -0,0 +1,12 @@
+package model
+
+//go:generate generator
+
+// CoderJunitTestResult is the result of a set of JUnit tests written by the
+// coder for some specs ran against code written by the coduno team for said
+// specs. Besides the results of the test, it also encapsulates wether it was
+// supposed to fail in order to assess the correctness of the unit tests.
+type CoderJunitTestResult struct {
+	JunitTestResult
+	ExpectedFailure bool
```

Comment:
`FailureExpected` would read more natural than `ExpectedFailure`, coming from "Failure is expected." and `isFailureExpected`. But I'd rather call this `ShouldFail`, because "It should fail." sounds more like a definition than "Failure is expected." which also reminds of "Is failure expected?" thus indicating that an answer to that question is stored, rather defining that this should fail.